database_service/database_connector.go

```diff
@@ -0,0 +1,30 @@
+package database_service
+
+import (
+	"strconv"
+
+	"github.com/leechongyan/Studtor_backend/authentication_service/models"
+	"github.com/leechongyan/Studtor_backend/helpers"
+	"github.com/spf13/viper"
+)
+
+var CurrentDatabaseConnector DatabaseConnector
+
+type DatabaseConnector interface {
+	// expose all the possible database api
+	SaveUser(user models.User) (err *helpers.RequestError)
+	GetUser(email string) (user models.User, err *helpers.RequestError)
+	GetAllCourses(from string, size int) (courses []string, err *helpers.RequestError)
+	GetAllTutors(from string, size int) (tutors []string, err *helpers.RequestError)
```

Comment:
why are db errors throwing http errors

Reply:
http errors are thrown as internal server error as we are ultimately tracking the error code for return to the user, thats why it is being used

Reply:
wouldn't it be better for the handler to decide what http code to show to the user based on the _type_ of error thrown internally? 

if a user cannot be found but the error is due to a timeout, we would return a 503; however, if the internal error is due to not found, we would return 404

doing this restricts our possibility right off the bat and the db layer now controls the request layer, which seems odd

Reply:
i feel that database should still deal with the error, ultimately we can still edit the error code within the database and by pushing up the error code to request layer, it just brings more confusion and switch statements. 503 and 404 can still be raised by the database at their layer

Reply:
the db throws _database_ related errors - if the db times out, that's a db issue ; it can throw whatever it wants at that layer. however, the concerns of the database should not matter because there might be other errors/more information thrown. 

the handler is the final layer between teh FE/BE and it has more info to decide what kinds of error to return back to the user including what error was thrown by the db.